modules/catalog/src/main/scala/lucuma/catalog/csv/TargetImport.scala

```diff
@@ -112,14 +114,28 @@ object TargetImport:
           .leftMap(_ => new DecoderError(s"Invalid epoch value '$r'"))
     }
 
-  given angularVelocityOpt[A]: CellDecoder[Option[ProperMotion.AngularVelocityComponent[A]]] =
+  given pmRADecoder: CellDecoder[Option[ProperMotion.RA]] =
     CellDecoder.stringDecoder.emap { r =>
       val t = r.trim()
       if (t.isEmpty) Right(None)
       else
         t.parseBigDecimalOption
           .map(r =>
-            ProperMotion.AngularVelocityComponent.milliarcsecondsPerYear
+            ProperMotion.RA.milliarcsecondsPerYear
+              .reverseGet(r)
+              .some
+          )
+          .toRight(new DecoderError(r))
+    }
+
+  given pmDecDecoder: CellDecoder[Option[ProperMotion.Dec]] =
```

Comment:
Can we avoid the duplicated code? Maybe we can pass `Propermotion.X.milliarcsecondsPerYear.reverseGet` as a parameter to a function.

Reply:
I made a mistake on core making the function we need private. I'll change it there and we can make this generic